code/game/area/areas.dm

```diff
@@ -105,6 +105,12 @@
 
 	. = ..()
 
+/area/proc/set_lightswitch(var/state)
+	var/obj/machinery/light_switch/L = locate() in src
```

Comment:
As explained above, this should also set the lightswitch variable. There is no good reason for the light switches to be out of sync with the area's light state.
```suggestion
	lightswitch = state
	var/obj/machinery/light_switch/L = locate() in src
```